src/components/schedule-visualizer/robot-image-icon.tsx

```diff
@@ -12,44 +12,74 @@ type RobotImageIconProps = Omit<RobotProps, 'colorManager' | 'fleetName'> & {
   >;
 };
 
+/**
+ *
+ * @param color MUST be in hex notation without alpha channel. e.g. #123456
+ */
+function makeGradientShadow(
+  color: string,
+): React.FunctionComponent<SVGProps<SVGRadialGradientElement>> {
+  return (props: SVGProps<SVGRadialGradientElement>) => (
+    <radialGradient {...props}>
+      <stop offset="0%" stopColor={`${color}80`} />
+      <stop offset="70%" stopColor={`${color}40`} />
+      <stop offset="90%" stopColor={`${color}10`} />
+      <stop offset="100%" stopColor={`${color}00`} />
+    </radialGradient>
+  );
+}
+
 const RobotImageIcon = React.forwardRef(function(
   props: RobotImageIconProps,
   ref: React.Ref<SVGGElement>,
 ): React.ReactElement {
-  const { robot, footprint, iconPath, dispatchIconError, inConflict } = props;
-  const theme = useTheme();
+  const { footprint, iconPath, dispatchIconError, inConflict, colorManager } = props;
+
   // The default icon uses footprint as the radius, so we * 2 here because the width/height
   // is in a square. With the double size of the footprint, we achieved a similar
   // size to the robot default svg icon.
   const [imgIconWidth, imgIconHeigth] = useMemo(() => [footprint * 2, footprint * 2], [footprint]);
 
+  const componentId = React.useMemo(uniqueId, []);
+  const shadowId = React.useMemo(() => `RobotImageIcon-${componentId}-shadow`, [componentId]);
+  const conflictShadowId = React.useMemo(() => `RobotImageIcon-${componentId}-shadow-conflict`, [
+    componentId,
+  ]);
+
+  const Shadow = React.useMemo(() => makeGradientShadow('#000000'), []);
+  const ShadowConflict = React.useMemo(() => makeGradientShadow(colorManager.conflictHighlight), [
+    colorManager.conflictHighlight,
+  ]);
+
   return (
     <>
       {!!iconPath && (
-        <g transform={`translate(${-footprint} ${-footprint})`}>
-          <filter id={`${robot.name}-shadow`} x="-20%" y="-20%" width="140%" height="140%">
-            <feDropShadow
-              dx="0"
-              dy="0"
-              stdDeviation={footprint * 0.15}
-              floodColor={inConflict ? theme.palette.error.main : theme.palette.common.black}
-            />
-          </filter>
-          <image
-            href={iconPath}
-            height={imgIconHeigth}
-            width={imgIconWidth}
-            filter={`url(#${robot.name}-shadow)`}
-            onError={error => {
-              console.error(
-                'An error occurred while loading the image. Using the default image.',
-                error,
-              );
-              return dispatchIconError(previousVal => {
-                return { ...previousVal, error: true };
-              });
-            }}
+        <g>
+          <defs>
+            <Shadow id={shadowId} />
+            <ShadowConflict id={conflictShadowId} />
+          </defs>
+          <circle
+            id="shadow"
+            r={footprint * 1.3}
+            fill={inConflict ? `url(#${conflictShadowId})` : `url(#${shadowId})`}
```

Comment:
Just to open a discussion, do you think this could be a separate component that handles the unique id logic and receives a `makeGradientShadow` function, `colorManager`,  `inConflict`, and `footprint` because the same logic it's been used in `robot-image-icon` and `robot-default-icon`. 

Since this is a solution that improves the performance, we will tend to replicate this solution for components that may have shadows and icons (just to give an example: dispensers), and in my opinion, it would be nice to have this logic in a separate component.

Reply:
The main problem is that the shadow is different for each of them. `makeGradientShadow` itself is a higher order component, I thought it would be more "react-ful" to let the parent render the component instead of passing in a HOC as a prop.

A problem with wrapping this logic in the component is that it can't be reuse for waypoints/dispenser anyway as they do not have `colorManager`, `inConflict`, `footprint` etc.

Reply:
> The main problem is that the shadow is different for each of them. `makeGradientShadow` itself is a higher order component, I thought it would be more "react-ful" to let the parent render the component instead of passing in a HOC as a prop.

Yeah, you're right, it's not the best idea to pass the HOC as a prop, we can keep it that way. 

> A problem with wrapping this logic in the component is that it can't be reuse for waypoints/dispenser anyway as they do not have `colorManager`, `inConflict`, `footprint` etc.

Yeah, sorry I think I expressed myself wrong, I was aware of this. But don't worry this stops making sense after your first comment.